### ISSUE TITLE
get rid of forked ChatKit

### ISSUE THREAD

**Issue by mahibi** [nextcloud/talk-android#1552]
in the past the fork https://github.com/nextcloud/ChatKit was created from https://github.com/stfalcon-studio/ChatKit
The modifications and the useage of https://github.com/nextcloud/ChatKit are too complicated.

The best solution would be to implement the needed designs etc on our own so we are more flexible.
Or at least we should use some well maintained lib (possibly switch back to original https://github.com/stfalcon-studio/ChatKit)

**Comment by jld3103** [comment]
The upstream ChatKit hasn't been updated in 1.5 years, we should probably not use it and just switch away from ChatKit completely.

**Comment by tobiasKaminsky** [comment]
Great idea, but which…?
I searched a bit, but did not find a good alternative…

**Comment by jld3103** [comment]
I think the plan was to build it all on our own, but that is a lot of work...

**Comment by andyxheli** [comment]
Would anything from this link help ?

https://android.libhunt.com/chatkit-alternatives

**Comment by jld3103** [comment]
All the suggestions are not related to chat ui libraries...

**Comment by AndyScherzinger** [comment]
None of the libs in the link provide anything related to a chat view 😔

**Comment by AndyScherzinger** [comment]
@jld3103 yeah, I am unsure if completely building it on our own is worth it or if looking into how to upgrade to their latest version/master would be possible since the lib actually offers some thing we want in their latest version like image or map chat bubbles.

**Comment by jld3103** [comment]
Forking the latest version and then building on top of that sounds good. Some initial work to update it, but a lot less work implementing the basic stuff.

**Comment by AndyScherzinger** [comment]
I quickly talked about it today mentioning this also to @mahibi
Issue is that it is also meshed together with the fresco fork I think. I tried it a while ago but failed due to time constraints on my side.

**Comment by jld3103** [comment]
I looked at the forked ChatKit and Fresco isn't used as a dependency in it. Not sure why everyone thinks this is the case, maybe it's just used together.

**Comment by mahibi** [comment]
yes it's just used together. didn't have a closer look now what exactly is related.
but regarding fresco: there is the need anyway to get rid of the forked fresco because of https://github.com/nextcloud/talk-android/issues/2376 (@timkrueger wants to work on this next i think..)
so maybe we should first try to replace fresco and then chatkit.

**Comment by timkrueger** [comment]
>but regarding fresco: maybe there is the need anyway to get rid of the forked fresco because of #2376 (@timkrueger wants to work on this next i think..) so maybe we should first try to replace fresco and then chatkit.
Yes. I'm currently replace Fresco with coil. By the way I try  to centralize some heavily distributed code to load avatar images.
https://github.com/nextcloud/talk-android/pull/2449

**Comment by mahibi** [comment]
once chatkit is replaced/updated we should switch to emoji2 (because newer emojis are not correctly shown atm)
https://developer.android.com/jetpack/androidx/releases/emoji2


However only the forked chatKit from nextcloud implemented the usage of emojis...:
https://github.com/nextcloud-deps/ChatKit/commit/1cdff9b81861e072ab7ec473532b1bf8714f794c#diff-f740b02064d567b4c1f12acbd1163a07c8a9e6aa296aeb628783514c0693b5c4

**Comment by AndyScherzinger** [comment]
@mahibi so for emojis then there is no dependency because we could still decide to replace it even in the current forked version of chat kit, no?

**Comment by mahibi** [comment]
> @mahibi so for emojis then there is no dependency because we could still decide to replace it even in the current forked version of chat kit, no?

yes. we can bump to emoji2 in the forked chatkit and in the talk app to avoid the version clash. We should do this when https://github.com/nextcloud/talk-android/pull/2449 is merged..

**Comment by mahibi** [comment]
just for the record: ChatKit is driving me totally crazy again so +1 to remove it!!

Replacing it with own code using Jetpack Compose would be worth it in my opinion.
